### PR TITLE
Privacy Center Additional Customizations

### LINKED ISSUE #3399: Ability to Customize Favicon on Privacy Center
### Is your feature request related to a specific problem?

Currently the Favicon for the Fides Privacy Center is not configurable and statically set to the Fides square icon. 

### Describe the solution you'd like

We need a way to pass in a favicon path in the privacy center configuration. It should support the same references as the logo field (e.g. file name or URL)

### Additional context

_Add any other context or screenshots about the feature request here._

### LINKED ISSUE #4091: Privacy Center Styling and Function Enhancements
### Is your feature request related to a specific problem?

When having the privacy center in a production environment, it is important for me to have the logo link back to my site. This will help with navigation for other users in lieu of being forced to backtrack via the browser.

Additionally, I want to be able to link to my privacy policy in a clear way from the home page of the privacy center. Currently, the only place this is found in the consent overlay.

This work should be packaged up with #3399 as a single group of styling/function changes

_update!:_ #3399 is actually already complete and can be closed

### Describe the solution you'd like

A simple way to configure the privacy center until an improved editing scheme is implemented

### Describe alternatives you've considered, if any

Delaying until a better experience exists, i.e. WYSIWYG 

### Additional context

Some of these changes are being tracked as critical to the use case of Fides as a privacy center

### PR DESCRIPTION
**Pull request by SteveDMurphy**
This makes the Logo it's own component and allows it to be treated as a link if a logo_url is supplied

Closes #3399
Closes #4091 

### Description Of Changes

This addresses some functionality changes for clients to be able to:
* Have the logo also link back to their site
* Link the privacy policy from the main privacy center (instead of just hidden in consent modal)
* ~favicon option~ already done!

### Code Changes

* [x] Move the image to a new component, `Logo`
* [x] Optionally link back to the site from the privacy center if a url is provided (for backwards compatability)
* [x] Between all of the text on the main privacy center page and the tiles, add a link to the privacy policy if config options supplied (this could be done much differently, and even repurpose the provided URL if needed from the modal. Thought separate might be better for this use case however.)

### Steps to Confirm

* [ ] `cd clients/privacy-center/`
* [ ] include the following config options in `~/config/config.json`
```
  {...
  "privacy_policy_url": "https://privacy.ethyca.com/",
  "privacy_policy_url_text": "Privacy Policy",
  "privacy_policy_visible": true,
  "logo_path": "https://cdn.ethyca.com/fides-cloud/ethyca/logo.svg",
  "logo_url": "https://www.ethyca.com/",
  ...}
  ```
* [ ] `turbo run dev`
* [ ] Validate the logo navigates away from the privacy center
* [ ] Validate the privacy policy link exists and opens in a new tab

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [x] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: clients/privacy-center/components/Logo.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Image } from "@fidesui/react";
+import React from "react";
+
+type LogoProps = {
+  logo_path: string;
+};
+
+const Logo: React.FC<LogoProps> = ({ logo_path }) => (
+  <Image
```
Comment (allisonking) [inline clients/privacy-center/components/Logo.tsx:9]: it might be nice to have the `Logo` component itself take an optional link, and let it be responsible for determining if it renders as a link logo or a regular logo. that would be useful if we wanted this logo with linking ability elsewhere, and it'd keep `Layout.tsx` a little cleaner. just a thought, certainly not blocking!

it could look something like...

```tsx
// Layout.tsx
<Logo href={config.logo_url} src={config.logo_path} /> // in general I like to name props after similar props (`a` takes an `href`, and `Image` takes a `src`)

// Logo.tsx
type LogoProps = {
  src: string;
  href?: string;
}

// basically what you currently have for Logo
const LogoImage = ({src}:Pick<LogoProps, 'src'>) => (
  <Image src={src} ... />
)

// simple wrapper around LogoImage
const Logo = ({src, href}:LogoProps) => {
  
{ href? (
      <a href={href}>
        <LogoImage src={src} />
      </a>
      ) : (
        <LogoImage src={src} />
      )}
}

export default Logo;
```

**Review by allisonking — APPROVED** [review]
works great, nice job Steve! left a few suggestions but they're not blocking, and are probably more on the opinion side :)

File: clients/privacy-center/components/Layout.tsx [hunk 2]
```diff
@@ -27,13 +28,13 @@ const Layout: React.FC<LayoutProps> = ({ children }) => {
...
         >
-          <Image
-            src={config.logo_path}
-            margin="8px"
-            height="68px"
-            alt="Logo"
-            data-testid="logo"
-          />
+          {config.logo_url ? (
+            <a href={config.logo_url}>
```
[the remark this answers is not in the document]
Comment (allisonking) [reply clients/privacy-center/components/Layout.tsx:32]: it can be! I think they render out to the same thing, but you're right that in general we prefer the `@fidesui` ones since they are easier to style

File: clients/privacy-center/pages/index.tsx [hunk 3]
```diff
@@ -187,6 +188,13 @@ const Home: NextPage = () => {
               {paragraph}
             </Text>
           ))}
+          {config.privacy_policy_visible &&
+          config.privacy_policy_url &&
+          config.privacy_policy_url_text ? (
+            <Link href={config.privacy_policy_url} isExternal>
+              {config.privacy_policy_url_text}
+            </Link>
+          ) : null}
```
Comment (NevilleS) [inline clients/privacy-center/pages/index.tsx:197]: Why also include `privacy_policy_visible`, shouldn't we just best infer that from the other configurable params?
```
config.privacy_policy_url && config.privacy_policy_url_text ? ( ... )
```

Comment (NevilleS) [reply clients/privacy-center/pages/index.tsx:197]: Additionally, some design feedback: this "Privacy Policy" link just feels... orphaned and out of place. Can we make it more intuitive? Maybe it feels better if we move this link below the action cards?

image


Let's get some design feedback and ship it!

**Review by NevilleS — CHANGES_REQUESTED** [review]
I did some quick local testing of this:
1. Edited the sample project (Cookie House) privacy center [config.json](https://github.com/ethyca/fides/blob/02f109c42364a3f041c09aa4dd416e4291cc2887/src/fides/data/sample_project/privacy_center/config/config.json) as:
```
{
  "title": "Take control of your data",
  "description": "When you use our services, you’re trusting us with your information. We understand this is a big responsibility and work hard to protect your information and put you in control.",
  "description_subtext": [],
  "addendum": [],
  "logo_path": "https://ethyca-cdn-harpocrates.s3.amazonaws.com/assets/cookiehouse_assets/logo.svg",
  "logo_url": "https://ethyca.com",
  "privacy_policy_url": "https://privacy.ethyca.com/",
  "privacy_policy_url_text": "Privacy Policy",
  "privacy_policy_visible": true,
  "actions": [
    {
      "policy_key": "default_access_policy",
...
```
2. Ran `nox -s "fides_env(test)"`
3. Opened up the Cookie House Privacy Center on http://localhost:3001/ and confirmed both the logo URL link and the Privacy Policy link:
image

So in other words: this works as advertised! However, the styling of the "Privacy Policy" link feels a bit out of place, I think we should try to polish that a bit.

File: clients/privacy-center/components/Logo.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Image } from "@fidesui/react";
+import React from "react";
+
+type LogoProps = {
+  logo_path: string;
```
[the remark this answers is not in the document]
Comment (SteveDMurphy) [reply clients/privacy-center/components/Logo.tsx:5]: thanks for this!! Will fix and get my casing squared away :)

File: clients/privacy-center/pages/index.tsx (hunk 3 above)
Comment (NevilleS) [reply clients/privacy-center/pages/index.tsx]: Pushed an update to this:
image

I think this is solid now so I'd say ship it

**Review by NevilleS — APPROVED** [review]
Thanks @SteveDMurphy! I pushed some minor styling improvements to the privacy policy link  and tidied my own nit, I'm happy enough with this one now 👍

**Comment by NevilleS** [comment]
Pushed a few minor updates, this is the look & feel now:
image

File: clients/privacy-center/components/Logo.tsx (hunk 1 above)
Comment (SteveDMurphy) [reply clients/privacy-center/components/Logo.tsx:9]: I really like this @allisonking - thank you!!

File: clients/privacy-center/components/Layout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (SteveDMurphy) [reply clients/privacy-center/components/Layout.tsx:32]: Thanks both! I think I started with that but was struggling to get it to function, so found `a` worked and leaned into it :) will fix!

**Comment by SteveDMurphy** [comment]
> Thanks @SteveDMurphy! I pushed some minor styling improvements to the privacy policy link and tidied my own nit, I'm happy enough with this one now 👍

Thank you a ton @NevilleS 🙌🏽

File: clients/privacy-center/components/Logo.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (SteveDMurphy) [reply clients/privacy-center/components/Logo.tsx]: resolved in https://github.com/ethyca/fides/pull/4090/commits/e486399aec3270808f09d162e0c955d753d6c764

File: clients/privacy-center/components/Logo.tsx (hunk 1 above)
Comment (SteveDMurphy) [reply clients/privacy-center/components/Logo.tsx]: resolved in https://github.com/ethyca/fides/pull/4090/commits/e486399aec3270808f09d162e0c955d753d6c764

File: clients/privacy-center/pages/index.tsx (hunk 3 above)
Comment (SteveDMurphy) [reply clients/privacy-center/pages/index.tsx]: Thanks for cleaning up that initial ask! This is much sharper ⭐

Comment (SteveDMurphy) [reply clients/privacy-center/pages/index.tsx]: Resolved in https://github.com/ethyca/fides/pull/4090/commits/82a2f7e90d57342210fde580320ab9f8acd8b07d

File: clients/privacy-center/components/Layout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (SteveDMurphy) [reply clients/privacy-center/components/Layout.tsx]: Resolved in https://github.com/ethyca/fides/pull/4090/commits/e486399aec3270808f09d162e0c955d753d6c764
